### ISSUE TITLE
Serialization of eosio name does not allow an empty string

### ISSUE THREAD

**Issue by fabian-emilius** [EOSIO/eosjs#801]
**Version of EOSJS**
21.0.3

**Describe the bug**
https://github.com/EOSIO/eosjs/blob/1cf9bf53d21d3f461f61970d3295ee1fc2ba56d1/src/eosjs-serialize.ts#L323
pushName() tests the name for a min length of 1 but the empty string "" should also be a valid eosio name (should be value 0)

**To Reproduce**
The following code fails
```javascript
const {SerialBuffer} = require("eosjs/dist/eosjs-serialize");

const buf = new SerialBuffer();
buf.pushName("");
console.log(buf.asUint8Array())
```
Expected Output
```
Uint8Array(8) [
  0, 0, 0, 0,
  0, 0, 0, 0
]
```

**Comment by adeelnaqvi172** [comment]
just used '...' this, 
where do u dont want to put any data but Serialization you error

**Comment by bradlhart** [comment]
This issue has since been changed in eosjs and the fix is within the edge release on NPM, but there is a workaround for the latest release: you can set the parent to '...' (or any number of periods as long as its within 13) and the serializer will resolve that to '' without causing the error.

**Comment by fabian-emilius** [comment]
Yes I already did that but wanted to report it anyway. Thanks for the answers!

**Comment by canab** [comment]
We cannot move further from eosjs@20.0.0 because of this bug.
Hundreds of our tests fail and there is no simple way to fix it.

**Comment by bradlhart** [comment]
This should be resolved with the v21.0.4 patch version.  Please let us know if there are any persistent problems regarding this issue to reopen.
